build_tools/scripts/bisect/install_packages_for_commit.py

```diff
@@ -0,0 +1,196 @@
+    download_artifacts_args = [
+        "gh",
+        "run",
+        "download",
+        "--repo",
+        f"{OWNER}/{REPO}",
+        str(run_id),
+        "--name",
+        "linux_x86_64_release_packages",
+        "--dir",
+        str(dir),
+    ]
+    print(
+        f"Running command to download artifacts:\n  {' '.join(download_artifacts_args)}"
+    )
+    subprocess.check_call(download_artifacts_args)
+
+
+def install_packages_from_directory(dir: Path, python_interpreter: str):
+    # Setup venv.
+    venv_dir = dir / ".venv"
+    print(f"Creating venv at '{venv_dir}'")
+    subprocess.check_call([python_interpreter, "-m", "venv", str(venv_dir)])
+    venv_python_interpreter = str(venv_dir / "bin" / "python")
+    subprocess.check_call(
+        [venv_python_interpreter, "-m", "pip", "install", "--upgrade", "pip", "--quiet"]
+    )
+
+    # Install common deps.
+    install_deps_args = [
+        venv_python_interpreter,
+        "-m",
+        "pip",
+        "install",
+        "--quiet",
+        "numpy",
+        "sympy",
+    ]
+    print("")
+    print(f"Running command to install dependencies:\n  {' '.join(install_deps_args)}")
+    subprocess.check_call(install_deps_args)
+
+    # Install each .whl in the directory.
+    # NOTE: this will fail if the Python interpreter is not the same version
+    # as the packages or if packages are ever built for multiple Python
+    # versions.
+    # TODO(scotttodd): Make this robust or at least log a better error message.
+    whl_files = list(dir.glob("*.whl"))
+    for file in whl_files:
+        install_package_args = [
+            venv_python_interpreter,
+            "-m",
+            "pip",
+            "install",
+            "--quiet",
+            str(file),
+        ]
+        print(
+            f"Running command to install package:\n  {' '.join(install_package_args)}"
+        )
+        subprocess.check_call(install_package_args)
+
+    # Log which packages are installed.
+    print("")
+    print(f"Checking packages with 'pip freeze':")
+    subprocess.check_call([venv_python_interpreter, "-m", "pip", "freeze"])
+
+
+def main(args):
+    print("------------------------------------------------------------------")
+    print(f"Installing packages for ref: {args.ref}")
+    print(f"  Using base working directory : '{args.work_dir}'")
+    Path.mkdir(args.work_dir, parents=True, exist_ok=True)
+
+    print("")
+    latest_run_id = get_latest_workflow_run_id_for_ref(args.ref)
+    artifacts_dir = args.work_dir / str(args.ref)
+    Path.mkdir(artifacts_dir, exist_ok=True)
+
+    existing_files = list(artifacts_dir.glob("*.whl"))
+    if existing_files:
+        print("Found cached .whl files in artifacts dir, skipping download")
+    else:
+        download_artifacts_for_run_id(latest_run_id, artifacts_dir)
+
+    install_packages_from_directory(
+        artifacts_dir, python_interpreter=args.python_interpreter
+    )
```

Comment:
Oh, we could probably reuse the https://github.com/iree-org/iree/blob/main/build_tools/pkgci/setup_venv.py script here... that has code to install packages matching the current system: https://github.com/iree-org/iree/blob/ad4cf1a588dc5e05122e533260072612ef516a77/build_tools/pkgci/setup_venv.py#L120-L129

Though it is limited to the wheel names in that file, so it won't allow bisecting before the point where `iree-compiler` was renamed to `iree-base-compiler` without some edits.

Reply:
Made that change. Deleted the somewhat duplicated code and moved files under `build_tools/pkgci/`.

Reply:
Oh I can go a step further and just teach `setup_venv.py` how to setup from a git ref instead of a workflow run. Doing that...